### PR TITLE
feat: add `SKIP_WORKFLOW_DURATION_ESTIMATION` env var. Fixes #7271

### LINKED ISSUE #7271: Add option to skip workflow duration estimation
# Summary

What change needs making?

I would like to have an option to disable workflow duration estimations in workflow-controller using setting in configMap.

# Use Cases

When would you use this?

Currently for workflows created from cronWorkflows, worfklowTemplates and clusterWorkflowTemplates controllers tries to estimate duration using previous workflow executions, at first it checks for them in kubernetes Api (existing workflow resources) in case there are none (like in my case `ttlStrategy.secondsAfterCompletion: 60` so wf's are moved to archive quickly) it checks the archive.
I've described problem with that in details here:
https://github.com/argoproj/argo-workflows/discussions/6670
In my case archive is postgres as RDS `db.t3.medium"` and I'm running ~25k workflows a day and I'm removing form archive workflows older than 20days. Controller often gets stuck waiting for response form database. It seems that this might be a problem with all argo-workflows setups with big number of daily wofklow execution. Estimation is not useful in every setup so having an ability to disable estimations would be useful feature to avoid this problems.

---

**Message from the maintainers**:

Love this enhancement proposal? Give it a 👍. We prioritise the proposals with the most 👍.

### PR DESCRIPTION
**Pull request by tooptoop4**
Fixes https://github.com/argoproj/argo-workflows/issues/7271
co-authored by @talebzeghmi in https://github.com/zillow/argo-workflows/pull/1

### CODE REVIEW

File: docs/environment-variables.md [hunk 1]
```diff
@@ -55,6 +55,7 @@ This document outlines environment variables that can be used to customize behav
...
 | `SEMAPHORE_NOTIFY_DELAY`                 | `time.Duration`     | `1s`                                                                                        | Tuning Delay when notifying semaphore waiters about availability in the semaphore                                                                                                                                                                                        |
 | `WATCH_CONTROLLER_SEMAPHORE_CONFIGMAPS` | `bool` | `true` | Whether to watch the Controller's ConfigMap and semaphore ConfigMaps for run-time changes. When disabled, the Controller will only read these ConfigMaps once and will have to be manually restarted to pick up new changes. |
+| `SKIP_WORKFLOW_DURATION_ESTIMATION` | `bool` | `false` | Whether to lookup resource usage from prior workflows as a basis for estimating resource usage on a new workflow. |
```
Comment (agilgur5) [inline docs/environment-variables.md:58]:
```suggestion
| `SKIP_WORKFLOW_DURATION_ESTIMATION` | `bool` | `false` | Whether to lookup resource usage from prior workflows to estimate usage for new workflows. |
```

- simplify, per [k8s style guide](https://kubernetes.io/docs/contribute/style/style-guide/#use-simple-and-direct-language)
- pluralize as it's for all new workflows

**Review by agilgur5 — APPROVED** [review]
Makes sense to me since this can be expensive

**Comment by agilgur5** [comment]
> co-authored by @talebzeghmi in [zillow#1](https://github.com/zillow/argo-workflows/pull/1)

I also added them to the commit message as co-author, sourcing the name + email from https://github.com/argoproj/argo-events/pull/3190

**Comment by talebzeghmi** [comment]
Thank you for merging this!! 🙏

**Pull request merged**